internal/controller/resource_manager.go

```diff
@@ -74,6 +100,9 @@ func (r *ResourceManager) DyanmicListAndWatchResources(ctx context.Context, canc
 		Watch:           true,
 		ResourceVersion: resourceListVersion,
 	}
+	// Prevent us from overwhelming K8 api
+	randomDuration := time.Duration(rand.Intn(5-1+1)+1) * time.Second
+	time.Sleep(randomDuration)
```

Comment:
Try using a proper rate limiter like e.g. https://pkg.go.dev/golang.org/x/time/rate instead of a hardcoded sleep.